ksql-metastore/src/main/java/io/confluent/ksql/metastore/MetaStoreImpl.java

```diff
@@ -83,8 +83,18 @@ public StructuredDataSource getSource(final String sourceName) {
   public Optional<StructuredDataSource> getSourceForTopic(final String ksqlTopicName) {
     return dataSources.values()
         .stream()
-        .filter(p -> p.source.getKsqlTopic().getName() != null
-            && p.source.getKsqlTopic().getName().equals(ksqlTopicName))
+        .filter(p -> p.source.getTopicName() != null
+            && p.source.getTopicName().equals(ksqlTopicName))
+        .map(sourceInfo -> sourceInfo.source)
+        .findFirst();
+  }
+
+  @Override
+  public Optional<StructuredDataSource> getSourceForKafkaTopic(final String kafkaTopicName) {
+    return dataSources.values()
+        .stream()
```

Comment:
nit: [not your original code] we can start by `.map(sourceInfo -> sourceInfo.source)` since we only use that anyway in the filter

Reply:
Nice eye. Will do in a follow-up.